pkg/didcomm/dispatcher/api.go

```diff
@@ -76,20 +76,20 @@ type OutboundCreator func(prov Provider) (Outbound, error)
 type DIDCommAction struct {
 	// DIDComm message
 	Message DIDCommMsg
-	// Callback function to be called by the consumer for further processing the message.
-	Callback Callback
+	// Continue function to be called by the consumer for further processing the message.
+	Continue Continue
+	// Fail invocation notifies the service that the consumer action event processing has failed or the consumer wants
+	// to stop the processing.
+	Fail Fail
 	// Properties contains different values from different protocols
 	Properties map[string]interface{}
 }
 
-// DIDCommCallback message type to pass service callback in go channels.
-type DIDCommCallback struct {
-	// Set the value in case of any error while processing the DIDComm message event by the consumer.
-	Err error
-}
+// Continue type to process service callbacks.
+type Continue func()
```

Comment:
Suggest we declare `DIDCommAction.Continue` a `func()` directly as opposed to defining this extra type

Reply:
In this case having type doesn't make much sense. Hence [removed](https://github.com/hyperledger/aries-framework-go/compare/cf0815053284baec1dece5480f30d8bd13ec3b36..45f8d6085ca35de12a950470fe29a49a0528996f#diff-6a4454074ff31ccb82bdbcc343255467L88).